Add a crypto menu

jmaslek opened the pull request:
In this PR, I added a crypto menu using the CoinGecko API.  

Tried to keep the calls similar to the loading/viewing of stock tickers.   Obviously a lot more work can be done., but this is a start.

jmaslek commented:
Okay I am lost.  Flake wants an extra blank line but black just gets rid of it.

DidierRLopes commented on gamestonk_terminal/cryptocurrency/README.md:
`````diff
@@ -0,0 +1,42 @@
+````
+After loading the above 1 day dogecoin/eur.  Running view plots the data:
+
+![doge](#https://user-images.githubusercontent.com/18151143/112675579-b83bad00-8e3d-11eb-87d9-db8d17896403.png)
`````
The image isn't showing properly

DidierRLopes commented on gamestonk_terminal/cryptocurrency/crypto_controller.py:
```diff
@@ -0,0 +1,107 @@
+        print("")
+        print("   load          load cryptocurrency data")
+        print("   view          load and view cryptocurrency data")
+        print("   clear          load cryptocurrency data")
```
I think the descriptions are wrong here! And there's an extra space after  clear

jmaslek replied:
yup I am seeing that now

DidierRLopes commented on gamestonk_terminal/cryptocurrency/crypto_controller.py:
```diff
@@ -0,0 +1,107 @@
+    CHOICES = ["help", "q", "quit", "load", "view", "clear"]
+
+    def __init__(self):
+
```
""" Constructor"""

jmaslek replied:
like docstring?

DidierRLopes replied:
yupp

DidierRLopes commented on gamestonk_terminal/cryptocurrency/crypto_controller.py:
```diff
@@ -0,0 +1,107 @@
+            print("No coin selected. Use 'load' to load the coin you want to look at.")
+            print("")
+
+    def call_clear(self, _):
```
I had the clear command like this before, but then ended up using the same argparse logic so that the user:
1. Can use the -h flag
2. If they do "clear blablabla", we output that "blablabla" hasn't been recognized

jmaslek replied:
So I am not even sure if `clear` is necessary....

Thoughts on just not having the clear (since loading a new coin just updates the current name/df

DidierRLopes replied:
Sounds good to me yea!

DidierRLopes commented on gamestonk_terminal/cryptocurrency/crypto_controller.py:
```diff
@@ -0,0 +1,107 @@
+
+    def call_clear(self, _):
+        self.current_coin = None
+        self.current_df = pd.DataFrame()
```
Missing a new line here

DidierRLopes reviewed (commented):
Nice!! Apart from those tiny comments, two other things that can be updated:

README inside gamestonk_terminal with new features that you've added so far! So that new users will know about these before installing the program.

Main README adding yourself as a collaborator!

DidierRLopes commented on gamestonk_terminal/cryptocurrency/coin_api.py:
```diff
@@ -81,18 +81,35 @@ def load(l_args):
+        add_help=False,
+        prog="Crypto",
+        description="""
+                        Cryptocurrencies
```
View crypto price or so

jmaslek commented:
So I addressed those comments + added the args logic to `view`.  I also added my info to the readme.  

I am heading outta town for the weekend, so I will address the added features readme next week.  Which speaking of: were you thinking just adding something like an added features list maybe after the usage video?  Or actually add the basic usages?

jmaslek commented:
Also -- is it possible to change that little flair in the far left?  Like make the crypto menu have coins instead of the stars?

DidierRLopes commented:
> So I addressed those comments + added the args logic to `view`. I also added my info to the readme.
> 
> I am heading outta town for the weekend, so I will address the added features readme next week. Which speaking of: were you thinking just adding something like an added features list maybe after the usage video? Or actually add the basic usages?

Perfect. Thanks!

No worries man, enjoy your weekend!

I meant here:
<img width="947" alt="Captura de ecrã 2021-03-26, às 21 07 57" src="https://user-images.githubusercontent.com/25267873/112692901-96f7b280-8e77-11eb-8834-43dbbcd76ee9.png">

Yup, the coin flair would be a nice touch eheh

jmaslek commented:
Ah I was looking at the wrong readme....
